### ISSUE TITLE
Unable to Update - "files would be overwritten by merge"

### ISSUE THREAD

**Issue by Sqrlly** [oobabooga/text-generation-webui#4360]
### Describe the bug

Upon trying to run `update_windows.bat` I get the following error:

```
.\update_windows.bat
remote: Enumerating objects: 48, done.
remote: Counting objects: 100% (48/48), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 31 (delta 26), reused 27 (delta 22), pack-reused 0
Unpacking objects: 100% (31/31), 5.92 KiB | 0 bytes/s, done.
From https://github.com/oobabooga/text-generation-webui
   7a3f885..b818314  main                -> origin/main
 * [new tag]         snapshot-2023-10-22 -> snapshot-2023-10-22
Updating 7e6ff8d..b818314
error: Your local changes to the following files would be overwritten by merge:
        extensions/Training_PRO/readme.md
Please commit your changes or stash them before you merge.
Aborting
Created autostash: bb477de
Applied autostash.
Command '"F:\AI\text-generation-webui\installer_files\conda\condabin\conda.bat" activate "F:\AI\text-generation-webui\installer_files\env" >nul && git pull --autostash' failed with exit status code '1'. Exiting...

```
I have tried deleting the extensions/Training_PRO/readme.md file, but it recreates it and then raises the same error about it.

I have tried doing a git stash push on it, but it says "No local changes to save" and doing an update still generates the error.

I did the update to requirements.txt and that worked, taking about a half hour.  Running the update after that still generates the same error above.

It seems I never have any luck updating this software!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Go to the "text-generation-webui" folder and enter ".\update_windows.bat"

### Screenshot

(Text provided in description above.)

### Logs

```shell
As above:

remote: Enumerating objects: 48, done.
remote: Counting objects: 100% (48/48), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 31 (delta 26), reused 27 (delta 22), pack-reused 0
Unpacking objects: 100% (31/31), 5.92 KiB | 0 bytes/s, done.
From https://github.com/oobabooga/text-generation-webui
   7a3f885..b818314  main                -> origin/main
 * [new tag]         snapshot-2023-10-22 -> snapshot-2023-10-22
Updating 7e6ff8d..b818314
error: Your local changes to the following files would be overwritten by merge:
        extensions/Training_PRO/readme.md
Please commit your changes or stash them before you merge.
Aborting
Created autostash: bb477de
Applied autostash.
Command '"F:\AI\text-generation-webui\installer_files\conda\condabin\conda.bat" activate "F:\AI\text-generation-webui\installer_files\env" >nul && git pull --autostash' failed with exit status code '1'. Exiting...
```

### System Info

```shell
Windows 10 64 bit
NVIDIA GeForce RTX 4070 12GB VRAM
80GB RAM
```

**Comment by FartyPants** [comment]
NOOOOOOOO. not that silly readme.md again....

Try to delete the entire Training_PRO folder. This github thing is on my hit list.

**Comment by Sqrlly** [comment]
Heya!  Well deleting the entire folder does the same as deleting the file.  It recreates it all, then spits out the same error that the file it created is modified. :-P

**Comment by bombel28** [comment]
@FartyPants Would you please repair the folder selection for raw text files as well?

**Comment by xpgx1** [comment]
Had the same issue, "corrected" it with "git checkout HEAD^ extensions/Training_PRO/README.md" - which was my conflict as well as the CMD_FLAGS.txt. 

But now the updater cannot find a missing dependency, namely "https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl". So - what now? 

Seems like an error on my end, I know, but when manually browsing to the repository - there is indeed nothing there =)

**Comment by Sqrlly** [comment]
@xpgx1 Yeah in my original comment I mentioned successfully updating the requirements.txt too.  Fortunately mine still starts so I may hold off on the checkout you did until we learn some more of what's going on.  But that's good to know.

**Comment by FartyPants** [comment]
About a month ago I somehow uploaded readme.md to the merge while a README.md was already there and github totally freaked out. (I'm on windows, github is on linux) This has been fixed of course - unless you have old version where the readme (low letters) is still present...

I'd just nuke the whole thing and install WebUI as new install again. But that's my solution to everything.

**Comment by Sqrlly** [comment]
It was README.md at first.  When I deleted it it replaced it with readme.md.  And one time I think it did replace it with the uppercase one too.  So seems they're both there?  Currently it's the lowercase one.  It may possibly be a month since my last update.

And it gets tiring doing full 6+ hour installs every time I need to update...  Then however long getting everything back to where it was... :-P

**Comment by Sqrlly** [comment]
Okay, been still struggling with this, but I finally got some progress.  In a command prompt window I changed directory to "extensions/Training_PRO" and from the command line (not windows explorer) did a `rm README.md` then did a `git pull` while still in that folder.  It updated it without the complaint.  A directory listing showed the README.md restored.  Then I went back up to the "text-generation-webui" folder and so far successfully running the `update_windows.bat` script.  It's not given that same error message at least.  Now just to wait, but looks like it's progressing!

**Comment by Werve** [comment]
I encountered the same problem and based on the previous comments this is what worked for me:

Start cmd_windows.bat

git fetch
git reset
cd "extensions\Training_PRO"
rm README.md
git merge @{u}

start update_windows.bat
